recipes-omnect/initrdscripts/omnect-os-initramfs/grub-sh

```diff
@@ -74,7 +74,10 @@ get_fsck_status()
 
 
 rootblk_dev_generate_dev_omnect() {
-    local search_rootblk=("/dev/sda" "/dev/nvme0n1p")
+    # we assume there is only one internal harddrive device
+    # we assume /dev/sdb is usb when sata device is /dev/sda (no verification though)
+    # we assume /dev/sda is usb when nvme device exists (no verification though)
+    local search_rootblk=("/dev/sdb" "/dev/sda" "/dev/nvme0n1p")
```

Comment:
what if we do a flash-mode 1 and forget to unplug the USB stick upon first boot?
wouldn't that result in rootblk to be set (and stored) to be that of the USB stick?
or do I miss some magic somewhere that would prevent the above scenario?
at least I didn't find anything that would persist the root device already in flash-mode 1 after the target (destination) device got prepared.

Reply:
yes rootblk is stored on the usb stick then. depending of, if the device is e.g. an arrakis pico vs an arrakis, the device boots from usb or not.
what is the problem in this case? what do you want to prevent exactly?
if the usb-stick gets unplugged at some point and the device is booted again, rootblk will get searched and then persisted in bootloader env on the disk.